### ISSUE TITLE
Intellisense for Angular Service (missing types)

### ISSUE THREAD

**Issue by henrysachs** [aws-amplify/amplify-js#3189]
**Describe the bug**
When using the amplify service i have no chance to find the proper methods on the different modules.

**To Reproduce**
add amplify service to your app. Type in for example 
```
this.amplifyService.auth().
```
dont get any recommendations.  I think there are some types missing.
This could correlate with me using angular 8 rc, if so actions need to be taken to support the new angular version.

**Expected behavior**
get a list of all avaible functions

**Screenshots**
![expected](https://user-images.githubusercontent.com/17173951/56900593-7a3d7f00-6a96-11e9-9ed5-97d7cd83be9b.png)
![current](https://user-images.githubusercontent.com/17173951/56900594-7a3d7f00-6a96-11e9-9369-71995c4e64e1.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome

**Smartphone (please complete the following information):**

**Additional context**
my package.json

```
 "dependencies": {
    "@angular/animations": "~8.0.0-rc.1",
    "@angular/cdk": "~8.0.0-beta.2",
    "@angular/common": "~8.0.0-rc.1",
    "@angular/compiler": "~8.0.0-rc.1",
    "@angular/core": "~8.0.0-rc.1",
    "@angular/forms": "~8.0.0-rc.1",
    "@angular/material": "^8.0.0-beta.2",
    "@angular/platform-browser": "~8.0.0-rc.1",
    "@angular/platform-browser-dynamic": "~8.0.0-rc.1",
    "@angular/pwa": "^0.800.0-rc.1",
    "@angular/router": "~8.0.0-rc.1",
    "@angular/service-worker": "~8.0.0-rc.1",
    "@ngx-translate/core": "^11.0.1",
    "@ngx-translate/http-loader": "^4.0.0",
    "aws-amplify": "^1.1.26",
    "aws-amplify-angular": "^3.0.1",
    "core-js": "^3.0.1",
    "hammerjs": "^2.0.8",
    "rxjs": "~6.5.1",
    "rxjs-compat": "^6.5.1",
    "tslib": "^1.9.0",
    "zone.js": "~0.9.0"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.800.0-rc.1",
    "@angular/cli": "~8.0.0-rc.1",
    "@angular/compiler-cli": "~8.0.0-rc.1",
    "@angular/language-service": "~8.0.0-rc.1",
    "@types/jasmine": "~3.3.12",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~11.13.8",
    "codelyzer": "^5.0.0",
    "jasmine-core": "~3.4.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.1.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.2",
    "prettier": "1.17.0",
    "protractor": "~5.4.0",
    "ts-node": "~8.1.0",
    "tslint": "~5.11.0",
    "tslint-config-prettier": "^1.18.0",
    "typescript": "~3.4.5"
  }
```

just to clarify i get it for the auth,storage level but dont in the one deeper

**Comment by henrysachs** [comment]
Hey thanks for adding tags but i dont think this is a feature request because it was working in previous versions. :)

**Comment by manueliglesias** [comment]
Ah! Thanks for pointing that out, I've fixed the tags and we'll look into it 😃

**Comment by haverchuck** [comment]
@henrysachs The issue is that using those types in the AmplifyService required the importation of all Amplify category packages, which inflated the bundle size.  We had to untype these methods in the service in order support modularization (this is also related to the package needing to support Angular 5, which had less support for tree-shakable providers).   This is why we marked the update as a breaking change.

However, I certainly see that this is a problem - we can brainstorm a way to fill the gap here.
